### PR TITLE
Public garden show

### LINKED ISSUE #37: As a visitor, when I visit a PUBLIC Garden Show page, while not logged in, I see a the garden show page
Tests:

- [ ] Logged in user **cannot** access **private** garden that they do not own
- [ ] Logged in user **can** access **private** garden that they do own
- [ ] Logged in user **does not** see a link back to home page

- [ ] Unauthenticated user **does not**  see a link to login/register
- [ ] Logged in user **cannot** access **private** garden
- [ ] Unauthenticated user **can** access **public** garden
- [ ] Unauthenticated user sees a link back to home page
- [ ] Unauthenticated user sees a link to login/register

### PR DESCRIPTION
**Pull request by leahriffell**
closes #37 

**Functionality**
- User can only access a private garden show page only if they own the garden

**What was changed**
- Implemented webmock to block API calls to Rails BE

**Notes**
- Did not refactor call into service/facade design yet
- VCRs should be utilized for testing, but not yet implemented
  - One implemented, remove json fixture files

### CODE REVIEW

File: app/controllers/gardens_controller.rb [hunk 1]
```diff
@@ -1,11 +1,20 @@
...
-  def show
-    # GET "api/v1/gardens/params[:id]" to get single garden
 
-              # this code is used only for testing
-    @garden = Garden.new(current_user.gardens.first)
+  def show
+    # move conn, response, parse to service and facade once we better udnerstand api response structure
+    conn = Faraday.new("https://solar-garden-be.herokuapp.com/api/v1/gardens/#{params[:id]}")
+    response = conn.get
+    parsed = JSON.parse(response.body, symbolize_names: true)
```
Comment (LHJE) [inline app/controllers/gardens_controller.rb:9]: Clever!

File: spec/features/gardens/show_spec.rb [hunk 2]
```diff
@@ -1,20 +1,46 @@
...
+                                  data: []}}})
+
+
+    # change to vcr fixture testing eventually (once there's data available to actually call)
+    public_response = File.read('spec/fixtures/public_garden.json')
+    stub_request(:get, "https://solar-garden-be.herokuapp.com/api/v1/gardens/#{@public_garden.id}").to_return(status: 200, body: public_response)
+
+    private_response = File.read('spec/fixtures/private_garden.json')
+    stub_request(:get, "https://solar-garden-be.herokuapp.com/api/v1/gardens/#{@private_garden.id}").to_return(status: 200, body: private_response)
+  end
```
Comment (LHJE) [inline spec/features/gardens/show_spec.rb:40]: I'm glad you got all this to work!

File: spec/fixtures/private_garden.json [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+          },
+          "garden_plants": {
+              "data": []
+          },
+          "plants": {
+              "data": []
+          }
+      }
+  }
+}
```
Comment (LHJE) [inline spec/fixtures/private_garden.json:40]: Gosh all of this is great!

**Review by LHJE — APPROVED** [review]
So so good.  Great Job buddy

File: app/controllers/gardens_controller.rb (hunk 1 above)
Comment (dcoleman21) [reply app/controllers/gardens_controller.rb:9]: Nice mock here!

File: app/poros/garden.rb [hunk 4]
```diff
@@ -1,23 +1,24 @@
 class Garden
-
   attr_reader :id,
               :name,
               :latitude,
               :longitude,
               :description,
-              :private,
+              :is_private,
```
Comment (dcoleman21) [inline app/poros/garden.rb:7]: We can double check this from BE..

File: app/poros/garden.rb [hunk 5]
```diff
@@ -1,23 +1,24 @@
...
+              :user_ids
 
   def initialize(data)
     @id = data[:id]
     @name = data[:attributes][:name]
     @latitude = data[:attributes][:latitude]
     @longitude = data[:attributes][:longitude]
     @description = data[:attributes][:description]
-    @private = data[:attributes][:private]
+    @is_private = data[:attributes][:private]
```
Comment (dcoleman21) [inline app/poros/garden.rb:18]: We can double check this from BE.

File: spec/poros/garden_spec.rb [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+                    "data": []
+                },
+                "plants": {
+                    "data": []
+                }
+            }
+        }
+    }
+
+    garden = Garden.new(attr.deep_symbolize_keys[:data])
```
Comment (dcoleman21) [inline spec/poros/garden_spec.rb:46]: Nice testing here

**Review by dcoleman21 — COMMENTED** [review]
This looks great. Nice job with the webmock. We may have to just check on the 'private' vs 'is_private' coming from BE?

**Pull request merged**